react/react1/week3/to-do-list-3/src/components/to-do-list.jsx

```diff
@@ -0,0 +1,300 @@
+import { useState } from "react";
+import { useEffect } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import BorderBox from "./border-box";
+import "./css/to-do-list.css";
+
+function ToDoList() {
+  const [toDoTaskTxtBox, setToDoTask] = useState("");
+  const [deadlineDate, setDeadlineDate] = useState(new Date());
+  const [dateErrorMsg, setDateErrorMsg] = useState("");
+  const [taskNameErrorMsg, setTaskNameErrorMsg] = useState("");
+  const [toDoList, setToDoList] = useState([]);
+
+  function inputTextChangeHandler(e) {
+    setToDoTask(e.target.value.trim());
+  }
+
+  function dateChangeHandler(date) {
+    const currentDate = new Date();
+    if (date <= currentDate) {
+      console.log("past date");
+      setDateErrorMsg("Pls enter a future date");
+    } else {
+      setDeadlineDate(date);
+      console.log(date);
+      setDateErrorMsg("");
+    }
+  }
+
+  function AddClickHandler() {
+    if (toDoTaskTxtBox === "") {
+      setTaskNameErrorMsg("Pls enter task name");
+    } else {
+      setTaskNameErrorMsg("");
+      let maxId = 0;
+      for (let i = 0; i < toDoList.length; i++) {
+        if (toDoList[i].id > maxId) maxId = toDoList[i].id;
+      }
+      setToDoList((prevTaskList) => [
+        ...prevTaskList,
+        {
+          id: maxId + 1,
+          task: toDoTaskTxtBox,
+          deadLine: deadlineDate,
+          isCompleted: false,
+          isEditModeOn: false,
+        },
+      ]);
+    }
+  }
+
+  function DeleteTask(taskId) {
+    console.log("dlete " + taskId);
+    setToDoList(toDoList.filter((item) => item.id !== taskId));
+  }
+  function checkBoxStateChange(taskId) {
+    const taskArray = [...toDoList];
```

Comment:
you do not need to spread the toDoList, you already have the ToDoList array therefore this is unnecessary.